core/unit_test/hip/TestHIP_UnifiedMemory_ZeroMemset.cpp

```diff
@@ -19,7 +19,7 @@
 
 namespace Test {
 
-// On MI300a with ROCM <= 6.2.0, hipMemsetAsync was failing with an error when
+// On MI300a with ROCM <= 6.2.1, hipMemsetAsync was failing with an error when
```

Comment:
is this only for MI300A? It should be the same for older archs

Reply:
We technically in our semantics don't allow you to do that on anything other than APUs

Reply:
@crtrott made the convincing argument that we automatically select it to behave this way for MI300A to take mental load of users.